tests/test_device.py

```diff
@@ -247,14 +248,17 @@ class DMTest(PyverbsAPITestCase):
     """
     Test various functionalities of the DM class.
     """
+    def setUp(self):
+        super().setUp()
+        _, _, attr_ex = self.devices[0]
```

Comment:
@EdwardSro Any reason why the test needs to run only over the first device in the list?

Reply:
It doesn't "need" to run on the first device. But running on one device only would save testing time.
Do you use the option of running tests on all devices (by not passing --dev on purpose)?

Reply:
I wonder why not you guys use the --dev option? - because if a user would like to run the tests on all devices on the system this will be a bug...

Reply:
> I wonder why not you guys use the --dev option? - because if a user would like to run the tests on all devices on the system this will be a bug...

Agree, --dev sets to the correct device

Reply:
We do use --dev. But when the user doesn't use it, we choose an arbitrary device (which is the first device).
This is not the case just here. If you look at the API tests of test_cqex, test_mr, test_device, and others, we do the same thing.
Tests that inherit from RDMATestCase though, randomly select the device (if --dev was not passed).
So I have the following options in mind:
1. Keep things as is
2. Change tests inheriting from PyverbsAPITestCase to run on all devices
3. Change  PyverbsAPITestCase behaviour to randomly choose one device (same as RDMATestCase)

I would like to hear your opinions based on your usage.

Reply:
if --dev is not provided, the runner (upper logic) should chose default, internal test logic should assume that device is already chosen.

Reply:
That's what I meant in the 3rd option, which seems the most reasonable to me.  (I can change RDMATestCase to choose the first device by default instead of randomizing).
I will do it in a different PR, I need to apply this change on the remaining tests.